### PR TITLE
Enable running from GitHub Actions.

### LINKED ISSUE #5: TODO: Transition to using GitHub Actions completely (instead of Travis)

### PR DESCRIPTION
**Pull request by fredrikekre**
Fixes #5 

Better reviewed with white-space changes hidden: https://github.com/JuliaRegistries/RegistryCI.jl/pull/80/files?w=1

### CODE REVIEW

File: src/AutoMerge/public.jl [hunk 1]
```diff
@@ -1,64 +1,61 @@
...
-                authorized_authors::Vector{String},
-                #
-                enable_travis_api_builds::Bool = true,
-                enable_travis_cron_builds::Bool = true,
-                master_branch::String = "master",
-                master_branch_is_default_branch::Bool = true,
-                suggest_onepointzero::Bool = true)
-    registry_head = env["TRAVIS_BUILD_DIR"]::String
+function run(env = ENV,
+             cicfg::CIConfig=auto_detect_ci_service(;env=env);
```
Comment (DilumAluthge) [inline src/AutoMerge/public.jl]: Should this be `CIService` instead of `CIConfig`?

Comment (DilumAluthge) [reply src/AutoMerge/public.jl]: Let's find out.

File: src/AutoMerge/ciservice.jl [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+###############
+struct Travis <: CIService
+    enable_cron_builds::Bool
+    enable_api_builds::Bool
+    function Travis(; enable_cron_builds=true, enable_api_builds=true)
+        return new(enable_cron_builds, enable_api_builds)
+    end
+end
+function conditions_met_for_pr_build(cfg::Travis; env, kwargs...)
+    # TODO: Should also check the PR is against "master" or maybe thats done later?
```
Comment (DilumAluthge) [inline src/AutoMerge/ciservice.jl]: Can't hurt to check that now as well.

File: src/AutoMerge/public.jl (hunk 1 above)
Comment (DilumAluthge) [reply src/AutoMerge/public.jl]: The answer was yes.

**Comment by DilumAluthge** [comment]
So here's a question for you. When Travis runs a PR build, it doesn't checkout the feature branch, it checks out the "invisible" merge commit that GitHub creates, i.e.:
```
git clone --depth=99999999 https://github.com/$owner/$repo.git $owner/$repo
cd $owner/$repo
git fetch origin +refs/pull/$pull_request_number/merge:
git checkout -qf FETCH_HEAD
```

Does GitHub Actions automatically do that for us? If not, we should make sure to check out the merge commit before we begin.

**Comment by fredrikekre** [comment]
Yea, for the `pull_request` event I am pretty sure it does. At least `GITHUB_REF` is then set to `refs/pull/$(PR)/merge`.

**Comment by DilumAluthge** [comment]
Looks good to me. Want to just YOLO merge it and see what happens?

**Comment by DilumAluthge** [comment]
> > I guess you also need to add the GH Actions workflow files. Do those go in this repo? Or in the General registry repo?
> 
> In the General repo, its like `.travis.yml`

Cool. Once we get it working on General, it would be useful if we could put the contents of the workflow files somewhere inside this repo. Maybe in a folder called `templates` or something like that. That way, if someone wants to set up automerge on their own registry, they can just copy the workflow files from the `templates` folder.

**Pull request merged**